### PR TITLE
feat: trailingSlash

### PR DESCRIPTION
**Pull request by fdemir**
I didn't write a test for this feature, yet. I've created a new property in StartOptions called `route`. We can provide more functionality for routing in the future. 

Please leave some preview. Thanks in advance.

### CODE REVIEW

File: src/server/context.ts [hunk 1]
```diff
@@ -439,6 +451,10 @@ export class ServerContext {
     };
 
     for (const route of this.#routes) {
+      if (this.#routerOptions.trailingSlash) {
```
Comment (fdemir) [inline src/server/context.ts]: this will add a `/` to end of the all routes patterns.

File: src/server/context.ts [hunk 2]
```diff
@@ -571,6 +587,10 @@ const DEFAULT_RENDER_FN: RenderFunction = (_ctx, render) => {
   render();
 };
 
+const DEFAULT_ROUTER_OPTIONS: RouterOptions = {
+  trailingSlash: false,
```
Comment (fdemir) [inline src/server/context.ts:601]: disabled by default.

**Comment by fdemir** [comment]
May we consider adding this option to the documentation

**Review by lucacasonato — COMMENTED** [review]
This looks good. It'll need a test before I can land it though. Another test fixture, like `tests/fixture_error/` is probably in order.

File: src/server/types.ts [hunk 3]
```diff
@@ -6,8 +6,13 @@ import { InnerRenderFunction, RenderContext } from "./render.tsx";
 
 export type StartOptions = ServeInit & FreshOptions;
 
+export interface RouterOptions {
+  trailingSlash?: boolean;
```
Comment (iuioiua) [inline src/server/types.ts:10]: Can you please add the `@default` JSDoc directive and briefly document the purpose of this option?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Hi @fdemir, could you please resolve the merge conflicts and provide a brief explainer/example in the documentation? Once done, this should be ready to merge.

**Comment by iuioiua** [comment]
Hi @fdemir, are you still willing and able to pursue this PR?

**Comment by deer** [comment]
Hey @iuioiua, I picked this one up since it seemed to be missing documentation. Opened a new PR.

**Comment by iuioiua** [comment]
Closing in favour of #1291, which uses this PR as it's foundation. Thank you, @fdemir.

**Pull request closed without merging**